pgbar.coffee

```diff
@@ -52,11 +54,26 @@ class PgbarItem
       @extractor = (data) =>
         return parseInt(data.pgbar[@settings.field_name][@settings.delta])
 
+  selectTarget: (current) ->
+    t = 1
+    # copy the array
+    targets = @settings.targets.concat()
+    while targets.length > 0
+      t = targets.shift()
+      if (current * 100 / t) <= parseInt(@settings.threshold, 10)
+        return t
+    return t
+
   poll: ->
     registry = Drupal.behaviors.polling.registry
     callback = (data) =>
       to_abs = @extractor(data)
-      @animate(to_abs) if to_abs != @current
+      best_target = @selectTarget(to_abs)
+      if best_target != @settings.target
+        @settings.target = best_target
```

Comment:
I don’t think it’s a good idea to use the settings to store a value. I would change it so that `@selectTarget()` is directly called from within `@animate()`.

Reply:
Changed.